src/uthash.h

```diff
@@ -177,24 +193,45 @@ do {
 
 #define HASH_MAKE_TABLE(hh,head)                                                 \
 do {                                                                             \
+  int _mem_ok = 1;                                                               \
   (head)->hh.tbl = (UT_hash_table*)uthash_malloc(sizeof(UT_hash_table));         \
-  if (!(head)->hh.tbl) {                                                         \
+  if (!((head)->hh.tbl)) {                                                       \
+    if (HASH_NOMEM_OK) {                                                         \
+      break;                                                                     \
+    }                                                                            \
     uthash_fatal("out of memory");                                               \
```

Comment:
One concern I have is that "a hash write operation" might not be very well defined, in the mind of the client programmer. The model of "just check `hh.mem_failed` after every hash operation" is *very* uncomfortably close to "just check `errno` after every library operation", which is anathema to me. If the client could set

    #define uthash_kinda_fatal(x) do { log("uh-oh! hash out of memory!"); } while (0)

then I'd feel much better.